import_export/resources.py

```diff
@@ -520,7 +535,7 @@ def import_row(self, row, instance_loader, using_transactions=True, dry_run=Fals
                     # validate_instance(), where they can be combined with model
                     # instance validation errors if necessary
                     import_validation_errors = e.update_error_dict(import_validation_errors)
-                if self.skip_row(instance, original):
+                if not skip_diff and self.skip_row(instance, original):
```

Comment:
@matthewhegarty I don't understand why you are adding `not skip_diff` here, can you please explain?

Reply:
This is an optimisation because at this point (if `skip_diff` is True), the `original` var is not set.  

If we allow `skip_row()` to run then (assuming `skip_unchanged` is True) it will iterate over all model fields, and throw `AttributeError` for each when it tries to read a value from a `None` object.

I could potentially move the `if not skip_diff` check to inside `skip_row` which might be a bit cleaner.

Reply:
I now see what you mean, thanks.

So the problem is that setting `skip_unchanged` to `True` is not compatible with  settings `skip_diff` to `True` because we need `original` to diff it, which is not what you want because you asked to skip diffs operations.

I'd say that we should check about this situation early, like in the `__init__` method and fail with a descriptive message when both options are set to `True`.

It would be good to know what you and others think about this solution.

Reply:
I can add an `__init__` check if required.

However, I think since users have to opt in to `skip_unchanged`, I think it is ok to add a check in `skip_row()`:

```python
        if not self._meta.skip_unchanged or self._meta.skip_diff:
            return False
```
(I'll then remove the [other check](https://github.com/matthewhegarty/django-import-export/blob/skip-diff/import_export/resources.py#L538))

I will also document in `skip_row()` why this check is present and why the two flags are not compatible.

The other option is to always set `original`, but I don't like this because it will remove some of the performance improvement.

Thanks for reviewing!

Reply:
I think your proposal for `skip_row()` is a good solution, developers using both variables set to `True` should be able to understand the behaviour and identify the solution.

Let's wait for others to comment before you do any modification about this.

Thanks for the PR, I think it is a good first step into optimising import speed.

Reply:
I've updated the PR as discussed and added an extra test.